File: src/Implicit/ImplicitTan.cs
using System;

namespace TinkerWorX.AccidentalNoiseLibrary
{
    public sealed class ImplicitTan : ImplicitModuleBase
    {
        public ImplicitTan(ImplicitModuleBase source)
        {
            this.Source = source;
        }

        public ImplicitModuleBase Source { get; set; }

        public override double Get(double x, double y)
        {
            return Math.Tan(this.Source.Get(x, y));
        }

        public override double Get(double x, double y, double z)
        {
            return Math.Tan(this.Source.Get(x, y, z));
        }

        public override double Get(double x, double y, double z, double w)
        {
            return Math.Tan(this.Source.Get(x, y, z, w));
        }

        public override double Get(double x, double y, double z, double w, double u, double v)
        {
            return Math.Tan(this.Source.Get(x, y, z, w, u, v));
        }
    }
}

File: src/Implicit/ImplicitTan.java
package Implicit;

public final class ImplicitTan extends ImplicitModuleBase {
    public ImplicitTan(final ImplicitModuleBase source) {
        this.Source = source;
    }

    public ImplicitModuleBase Source;

    public ImplicitModuleBase getSource() {
        return this.Source;
    }

    public void setSource(final ImplicitModuleBase source) {
        this.Source = source;
    }

    @Override
    public double get(final double x, final double y) {
        return Math.tan(this.Source.get(x, y));
    }

    @Override
    public double get(final double x, final double y, final double z) {
        return Math.tan(this.Source.get(x, y, z));
    }

    @Override
    public double get(final double x, final double y, final double z, final double w) {
        return Math.tan(this.Source.get(x, y, z, w));
    }

    @Override
    public double get(final double x, final double y, final double z, final double w, final double u, final double v) {
        return Math.tan(this.Source.get(x, y, z, w, u, v));
    }
}
